src/Raven.Server/Documents/Indexes/VectorSearch/GenerateEmbeddings.cs

```diff
@@ -1,28 +1,60 @@
+#pragma warning disable SKEXP0070
 using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Text;
+using FastBertTokenizer;
+using Microsoft.ML.OnnxRuntime;
+using Microsoft.SemanticKernel.Connectors.Onnx;
 using Raven.Client.Documents.Indexes.Vector;
 using Raven.Client.Documents.Queries.Vector;
-using SmartComponents.LocalEmbeddings;
+using Raven.Server.Config;
 using Sparrow;
 using Sparrow.Server;
+using InvalidOperationException = System.InvalidOperationException;
 using VectorValue = Corax.Utils.VectorValue;
 
 namespace Raven.Server.Documents.Indexes.VectorSearch;
 
 public static class GenerateEmbeddings
 {
+    private static readonly ConstructorInfo BertOnnxTextEmbeddingGenerationServiceCtor;
+
     // Dimensions (buffer size) from internals of SmartComponents.
     private const int F32Size = 1536;
-    private static readonly LocalEmbedder Embedder = new();
+
+    private static SessionOptions OnnxSessionOptions;
+
+    internal static readonly Lazy<BertOnnxTextEmbeddingGenerationService> Embedder = new(CreateTextEmbeddingGenerationService);
+
+    static GenerateEmbeddings()
+    {
+        BertOnnxTextEmbeddingGenerationServiceCtor = typeof(BertOnnxTextEmbeddingGenerationService).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, [typeof(InferenceSession), typeof(BertTokenizer), typeof(int), typeof(BertOnnxOptions)]);
+        if (BertOnnxTextEmbeddingGenerationServiceCtor == null)
+            throw new InvalidOperationException($"Could not find constructor for {typeof(BertOnnxTextEmbeddingGenerationService)}.");
+    }
+
+    public static void Configure(RavenConfiguration configuration)
+    {
+        if (Embedder.IsValueCreated)
+            throw new InvalidOperationException("Embedder has already been initialized.");
+        
+        OnnxSessionOptions = new SessionOptions() { IntraOpNumThreads = configuration.Indexing.MaxNumberOfCoresForLocalEmbeddingsGeneration };
```

Comment:
should we also put the `InterOpNumThreads` here?

Reply:
for reference, more interesting options that we can add later if needed
https://onnxruntime.ai/docs/performance/tune-performance/threading.html